SAT/Z3Interfacing.hpp

```diff
@@ -197,10 +197,12 @@ class Z3Interfacing : public PrimitiveProofRecordingSATSolver
   Map<SortId, z3::sort> _sorts;
   struct Z3Hash {
     static unsigned hash(z3::func_decl const& c) { return c.hash(); }
+    static unsigned hash(z3::expr const& c) { return c.hash(); }
     static bool equals(z3::func_decl const& l, z3::func_decl const& r) { return z3::eq(l,r); }
+    static bool equals(z3::expr const& l, z3::expr const& r) { return z3::eq(l,r); }
```

Comment:
Originally I think we were hashing the bytes of a `z3::expr` via `Lib::Hash`...this seems better.